File: day-6/solution.cpp
class Solution {
public:
	bool go(vector<int>& furnitures,int value,int jump){
            if(!count(furnitures.begin(),furnitures.end(),value))return false;
    	if(value==furnitures[furnitures.size()-1])return true;
    	if(jump==0)return false;
    	if(go(furnitures,value+jump+1,jump+1))return true;
    	if(go(furnitures,value+jump,jump))return true;
    	if(go(furnitures,value+jump-1,jump-1))return true;
    	return false;
	}
    
	bool ThefloorIsLava(vector<int>& furnitures) {
    	for(int i=1;i<furnitures.size();i++){
        	if(furnitures[i]-furnitures[i-1]>i)return false;
    	}
    	return go(furnitures,furnitures[1],1);
	}
};

File: day-6/solution.py
class Solution:
	def ThefloorIsLava(self, furnitures):
    	"""
    	:type furnitures: List[int]
    	:rtype: bool
    	"""
    	dic = collections.defaultdict(set)
    	dic[0].add(0)
    	for i in range(len(furnitures)):
        	if furnitures[i] in dic:
            	for val in dic[furnitures[i]]:
                	if val > 0:
                    	dic[furnitures[i]+val].add(val)
                	if val > 1:
                    	dic[furnitures[i]+val-1].add(val-1)
                	dic[furnitures[i]+val+1].add(val+1)
    	return furnitures[-1] in dic
